### PR TITLE
Issue #2070 metric registry should create metadata copy

### PR DESCRIPTION
**Pull request by Arjun-Sharma1**

### CODE REVIEW

**Comment by Arjun-Sharma1** [comment]
#build

File: dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/impl/MetricRegistryImpl.java [hunk 1]
```diff
@@ -150,14 +150,16 @@ public MetricRegistryImpl() {
     @Override
     public <T extends Metric> T register(String name, T metric) throws IllegalArgumentException {
         // For MP Metrics 1.0, MetricType.from(Class in) does not support lambdas or proxy classes
-        return register(name, metric, new Metadata(name, from(metric)));
+        Metadata metadataCopy = new Metadata(name, from(metric));
```
Comment (raymondlam) [inline dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/impl/MetricRegistryImpl.java]: This is unnecessary

File: dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/impl/MetricRegistryImpl.java [hunk 2]
```diff
@@ -150,14 +150,16 @@ public MetricRegistryImpl() {
...
+        return register(name, metric, metadataCopy);
     }
 
     @Override
     public <T extends Metric> T register(String name, T metric, Metadata metadata) throws IllegalArgumentException {
 
         final Metric existing = metrics.putIfAbsent(name, metric);
-        this.metadata.putIfAbsent(name, metadata);
+        Metadata metadataCopy = new Metadata(metadata.getName(), metadata.getDisplayName(), metadata.getDescription(), metadata.getTypeRaw(), metadata.getUnit(), metadata.getTagsAsString());
```
Comment (raymondlam) [inline dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/impl/MetricRegistryImpl.java]: This method needs to be overridden in 1.1 to handle the `reusable` attribute

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
com.ibm.ws.app.manager_fat FAT tests - defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=250808 

com.ibm.ws.clientcontainer.cdi_fat FAT tests - defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=251612

com.ibm.ws.logstash.collector_fat FAT tests - defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=248642

com.ibm.ws.microprofile.faulttolerance.cdi_fat FAT tests	- defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=245322

com.ibm.ws.rest.api.discovery.batch_fat FAT tests - defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=251593

com.ibm.ws.security.javaeesec_fat FAT tests - defect already exists	
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=251368

com.ibm.ws.security.javaeesec_fat javacores - defect already exists
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=251215

**Comment by raymondlam** [comment]
Delivered with #2296

**Pull request closed without merging**